aten/src/ATen/native/cuda/SummaryOps.cu

```diff
@@ -31,16 +34,22 @@ namespace cuda {
  */
 enum class CUDAHistogramMemoryType { SHARED, MULTI_BLOCK, GLOBAL };
 namespace {
-  template<typename input_t, typename IndexType>
-  __device__ static IndexType getBin(input_t bVal, input_t minvalue, input_t maxvalue, int64_t nbins) {
-    IndexType bin = (int)((bVal - minvalue) * nbins / (maxvalue - minvalue));
-    // (only applicable for histc)
-    // while each bin is inclusive at the lower end and exclusive at the higher, i.e. [start, end)
-    // the last bin is inclusive at both, i.e. [start, end], in order to include maxvalue if exists
-    // therefore when bin == nbins, adjust bin to the last bin
-    if (bin == nbins) bin -= 1;
-    return bin;
-  }
+template <typename input_t, typename bounds_t, typename IndexType>
```

Comment:
do you need to string through another template argument, or could you have just declared `acc_type<input_t, true> minvalue`?

Reply:
To be honest, neither thing is aesthetically satisfying, to the point where I couldn't tell the difference. But in lieu of my own taste, I'll gladly go with yours. Thank you!

Reply:
I agree, neither is aesthetically pleasing, but I have a reflexive reaction to extra template argument "Oh, you are instantiating new templates" even though you aren't, and some code digging is required to make sure that no new instantiations happen.